ze/tracer_ze_helpers.include.c

```diff
@@ -952,7 +955,7 @@ void readEnergy(uint32_t deviceIdx, uint32_t domainIdx, uint64_t *ts_us, uint64_
 void readComputeE(uint32_t deviceIdx, computeEngineData *computeData ){
   if (!_sampling_engines_initialized) return;
   ze_result_t result;
-  for (uint32_t i = 0; i < _sampling_subDeviceCount; i++) {
+  for (uint32_t i = 0; i < _sampling_subDeviceCount[deviceIdx]; i++) {
```

Comment:
If a device has no subdevices, you will have 0 in `_sampling_subDeviceCount[deviceIdx]` and will not initialize the array.